src/iterable.cr

```diff
@@ -51,6 +51,20 @@ module Iterable(T)
     each.cons_pair
   end
 
+  # Same as `each.step(n)`.
+  #
+  # See also: `Iterator#step`.
+  def each_step(n : Int)
+    each.step(n)
+  end
+
+  # Same as `each.skip(offset).step(n)`.
+  #
+  # See also: `Iterator#step`.
+  def each_step(n : Int, *, offset : Int)
+    each.skip(offset < 0 ? offset % n : offset).step(n)
```

Comment:
What's with that negative offset? `offset % n` doesn't seem to make much practical sense?
Negative offsets are usually counted from the back, i.e. `-3` would be equivalent to `size - 3`.

Of course this doesn't work for arbitrary `Enumerable` or `Iterable`. So I figure the best thing would be to raise if `offset` is negative.

Regardless of what happens on negative offset, there should be a spec for that.

Reply:
I guess we could require `0 <= offset < n` for the time being, and it will still cover all possible uses; relaxing this condition isn't a breaking change, but the opposite direction is

Reply:
I have removed the negative offset behavior, now it will raise an `ArgumentError` on a negative offset. (offset of 0 is still valid)